### PR TITLE
Upgrade to TCA 1.4.2

### PR DESCRIPTION
**Pull request by kugel3**
## Description
Upgrade to TCA 1.4

### Notes
This also disables macro validation on CI. Please check that this is done in a suitable point in the CI script

## How to test
Everything should keep working as before. Navigation is most likely to break, it shouldn't affect smaller details.

## PR submission checklist
- [ ] I have tested account to account transfer flow and have confirmed that it works

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -189,7 +188,9 @@ jobs:
           brew install xcbeautify
 
       - name: "Unit tests"
-        run: bundle exec fastlane tests
+        run: |
+          defaults write com.apple.dt.Xcode IDESkipMacroFingerprintValidation -bool YES
```
Comment (Sajjon) [inline .github/workflows/ci.yml:192]: nice that it was this simple!

File: RadixWallet.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 2]
```diff
@@ -306,6 +306,15 @@
         "version" : "0.5.0"
       }
     },
+    {
+      "identity" : "swift-syntax",
```
Comment (Sajjon) [inline RadixWallet.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:310]: does this increase build time? if it does, we ought to bump timeout from 18 to.. 20 min?

File: RadixWallet/Features/DappInteractionFeature/Coordinator/DappInteractionFlow.swift [hunk 3]
```diff
@@ -146,6 +146,7 @@ struct DappInteractionFlow: Sendable, FeatureReducer {
 					.ifCaseLet(/MainState.chooseAccounts, action: /MainAction.chooseAccounts) {
 						AccountPermissionChooseAccounts()
 					}
+				EmptyReducer()
```
Comment (Sajjon) [inline RadixWallet/Features/DappInteractionFeature/Coordinator/DappInteractionFlow.swift:149]: intentional? is there some kind of contraint on only being able to do 3 `ifCaseLet`? Sounds arbitrary?

Hmm... should we not use Scope here btw??

Comment (kugel3) [reply RadixWallet/Features/DappInteractionFeature/Coordinator/DappInteractionFlow.swift:149]: The problem here is that type checking takes too long, with the new overloads.

**Review by CyonAlexRDX — APPROVED** [review]
LGTM

File: RadixWallet/Features/DappInteractionFeature/Coordinator/DappInteractionFlow.swift (hunk 3 above)
Comment (kugel3) [reply RadixWallet/Features/DappInteractionFeature/Coordinator/DappInteractionFlow.swift]: But yes, Scope also works, so I switched to the new CaseKeyPath version of Scope

**Pull request merged**